autotx/utils/agent/define_tasks.py

```diff
@@ -6,23 +6,111 @@
+                "context": [int] // The task_id that task execution depends on indicates which task_id execution result depends on which task execution depends. Eg. [0, 1] or null if no tasks need to be depended on.
                 "extra_information": Any extra information as string with description given by the user needed to execute the task, if applicable.
             }}]
         }}
+        Please note that you are only permitted to output json content.
 
         The specific tasks will be created based on the available agents role, goal and available tools:
         {agents_information}
-
-        IMPORTANT: After all tasks are executed, the prepared transactions will be sent to the Ethereum network.
+        
+        Please note: You can only output content in json format, and do not output any other content!
+        
+        You can refer to the following examples:
+        Goal: Send 1 ETH from the address 0x5d15311D760511d89cFad67404131cdc155E9FDB to the ENS domain vitalik.eth.
+        Output:
+        {{
+            "tasks": [{{
+                "task": "Check the balance of 1 ETH for the address 0x5d15311D760511d89cFad67404131cdc155E9FDB",
+                "agent": "send-tokens",
+                "expected_output": "A numeric value representing the balance of ETH in the account",
+                "context": null,
+                "extra_information": "Make sure the balance is at least 1 ETH to proceed with the transaction."
+            }}, {{
+                "task": "Prepare a transfer transaction of 1 ETH to the ENS domain vitalik.eth",
+                "agent": "send-tokens",
+                "expected_output": "A message confirming that the ETH transfer transaction has been prepared",
+                "context": [0],
+                "extra_information": "Use the output of the first task to ensure there is enough balance before proceeding."
+            }}]
+        }}
+        
+        Goal: Execute a swap from ETH to 100 USDC for the user's address 0x5d15311D760511d89cFad67404131cdc155E9FDB
+        Output:
+        {{
+            "tasks": [
+                {{
+                    "task": "Get current ETH balance of the user",
+                    "agent": "send-tokens",
+                    "expected_output": "Current ETH balance of 0x5d15311D760511d89cFad67404131cdc155E9FDB",
+                    "context": null,
+                    "extra_information": "Use Get ETH balance tool to check current ETH balance of user's address."
+                }},
+                {{
+                    "task": "Get current USDC balance of the user",
+                    "agent": "send-tokens",
+                    "expected_output": "Current USDC balance of 0x5d15311D760511d89cFad67404131cdc155E9FDB",
+                    "context": null,
+                    "extra_information": "Use Get ERC20 balance tool with USDC token symbol to check current USDC balance of user's address."
```

Comment:
since the action is swap ETH to USDC, this might not be necessary at all (it shouldn't care about the USDC balance)

we've noticed this in the tasks definition and removing this might improve the problem :thinking:

Reply:
This decomposition is done by gpt-4 and I will modify it as you said.